### PR TITLE
Help: Allow regular users to suggest regexes

### PR DESCRIPTION
**Pull request by mia-pi-git**
Approved by Help staff.

### CODE REVIEW

**Comment by mia-pi-git** [comment]
> I PMd you on Discord with review changes as you requested

(resolved)

**Comment by DaWoblefet** [comment]
I've been out of the loop with Help room filters, but why would we ever want regular users to be able to suggest regexes in some sort of formal manner like this? Are there really so many _regular users_ who are 1) technically skilled enough to write a proper regular expression and 2) can be trusted enough, to warrant a full-blown command? This seems very niche and if you trust someone enough to do those things, they should just be Help staff.

**Comment by mia-pi-git** [comment]
> I've been out of the loop with Help room filters, but why would we ever want regular users to be able to suggest regexes in some sort of formal manner like this? Are there really so many _regular users_ who are 1) technically skilled enough to write a proper regular expression and 2) can be trusted enough, to warrant a full-blown command? This seems very niche and if you trust someone enough to do those things, they should just be Help staff.

It doesn't actually allow regexes. It allows use of some symbols that are transformed into very simple regexes. You can only use actual regex when you're both dev staff and help mod. 
It comes down to like, &, !, and |, which i feel are rather self-explanatory.

File: server/chat-plugins/help.ts [hunk 1]
```diff
@@ -322,22 +383,36 @@ export const commands: ChatCommands = {
...
+				return this.errorReply(`You are banned from making suggestions to the Help filter.`);
+			}
 			if (!faq) return this.errorReply(`Invalid FAQ.`);
 			const regex = Answerer.stringRegex(target);
-			if (Answerer.queue.includes(target)) {
+			const entry = {
+				regexString: target,
+				userid: user.id,
+			};
+			if (Answerer.queue.includes(entry)) {
```
Comment (whalemer) [inline server/chat-plugins/help.ts:411]: this is an object, so it's checked by identity and this won't ever be true

**Comment by Zarel** [comment]
I guess if they want it...

**Pull request merged**